admin-dev/themes/new-theme/js/pages/customer/CustomerForm.ts

```diff
@@ -31,9 +31,54 @@ import ChangePasswordHandler from '../../components/change-password-handler';
  */
 export default class CustomerForm {
   constructor() {
+    // Watch password field change and strength indicator
     const passwordHandler = new ChangePasswordHandler(
       customerFormMap.passwordStrengthFeedbackContainer,
     );
     passwordHandler.watchPasswordStrength($(customerFormMap.passwordInput));
+
+    // Watch customer group checkbox change and if it was unchecked,
+    // update default group below if it's no longer in the list.
+    $('input[type="checkbox"][name="customer[group_ids][]"]').on('change', (event) => {
+      this.checkOrUpdateDefaultGroup($(event.currentTarget).is(':checked'));
+    });
+  }
+
+  private checkOrUpdateDefaultGroup(wasChecked: boolean): void {
+    // Get currently selected group ID
+    const currentDefaultGroup = Number($('#customer_default_group_id option:selected').val() as string);
```

Comment:
strange casting to string inside number :thinking:
Maybe type hinting would be enough (as I know its preferable)
```suggestion
    const currentDefaultGroup = Number(<string> $('#customer_default_group_id option:selected').val());
```